kernel/ata/src/lib.rs

```diff
@@ -685,6 +685,7 @@ impl BlockIo for AtaDrive {
 }
 impl KnownLength for AtaDrive {
 	fn len(&self) -> usize { self.block_size() * self.size_in_blocks() }
+	fn is_empty(&self) -> bool { self.block_size() * self.size_in_blocks() == 0 }
```

Comment:
```suggestion
	fn is_empty(&self) -> bool { self.len() == 0 }
```
It's silly to duplicate code for this, could you please also adjust all the other occurences?